### PR TITLE
Add support for getting env var selectors in docker

### LINKED ISSUE #1198: Proposal: support env selectors in the docker workload attestor
In my deployment, we are currently using the docker workload attestor. We have a system which runs docker containers in our fleet and attaches label at launch time which has been effective for workload attestation.

Other workloads in our deployment are managed by mesos. Mesos has support for "labels" as well as docker containers, but the labels are not set on the container they are only available from the mesos slave API. Further to this, we have many different ways of creating a mesos job (aurora, a custom scheduler, multiple orchestration platforms on top of each) and they do not agree about the labels in mesos at all. Additionally, for legacy reasons some of those systems cannot currently be changed.

The lowest common denominator available happens to be a few environment variables that all of our workloads currently depend on, therefore all of these orchestration platforms ensure they are set correctly.

We propose adding support for a env selector to the docker workload attestor. These env variables are the *initial values* attached to the container when it is created in the docker daemon, it is not mutable by the workload. If we instead extended the unix attestor to get data from `/proc/[pid]/environ` then it *would* be mutable by the workload, therefore we think it is best to use the docker workload attestor for this, not the unix workload attestor.

This doesn't change the trust model of the docker attestor, it will increase the number of selectors the agent cache needs to deal with but this will depend on the environment.

### PR DESCRIPTION
**Pull request by ZymoticB**
Fixes #1198

Signed-off-by: Tyler Dixon <tylerd@uber.com>

### CODE REVIEW

**Comment by ajessup** [comment]
Thanks for submitting Tyler! Mind updating the attestor docs too?

**Comment by ZymoticB** [comment]
> Thanks for submitting Tyler! Mind updating the attestor docs too?

Absolutely, thanks for the reminder

File: pkg/agent/plugin/workloadattestor/docker/docker_test.go [hunk 1]
```diff
@@ -20,40 +20,47 @@ import (
...
-			desc:                "single label",
+			desc:                "single label; single env",
 			mockContainerLabels: map[string]string{"this": "that"},
+			mockEnv:             []string{"VAR=val"},
 			requireResult: func(t *testing.T, res *workloadattestor.AttestResponse) {
-				require.Len(t, res.Selectors, 1)
+				require.Len(t, res.Selectors, 2)
 				require.Equal(t, "docker", res.Selectors[0].Type)
 				require.Equal(t, "label:this:that", res.Selectors[0].Value)
+				require.Equal(t, "docker", res.Selectors[0].Type)
```
Comment (azdagron) [inline pkg/agent/plugin/workloadattestor/docker/docker_test.go]: oops, `res.Selectors[1]`

File: pkg/agent/plugin/workloadattestor/docker/docker_test.go [hunk 2]
```diff
@@ -20,40 +20,47 @@ import (
...
+				require.Len(t, res.Selectors, 5)
+				expectedSelectors := []string{
+					"label:this:that",
+					"label:here:there",
+					"label:up:down",
+					"env:VAR=val",
+					"env:VAR2=val",
 				}
-				for _, selector := range res.Selectors {
+				for i, selector := range res.Selectors {
```
Comment (azdagron) [inline pkg/agent/plugin/workloadattestor/docker/docker_test.go]: I think the test is now subject to random failures due to map iteration order....

Comment (ZymoticB) [reply pkg/agent/plugin/workloadattestor/docker/docker_test.go]: 🤦‍♂ I guess @APTy had a good reason for the map :P.

Comment (ZymoticB) [reply pkg/agent/plugin/workloadattestor/docker/docker_test.go]:
```
go test . -count=100
ok  	github.com/spiffe/spire/pkg/agent/plugin/workloadattestor/docker	2.369s
```

**Pull request merged**